core/trino-main/src/main/java/io/trino/server/CoordinatorModule.java

```diff
@@ -341,6 +351,22 @@ protected void setup(Binder binder)
         binder.bind(ExecutorCleanup.class).in(Scopes.SINGLETON);
     }
 
+    @Provides
+    @Singleton
+    public static List<LowMemoryKiller> getLowMemoryKillers(@ForTask LowMemoryKiller lowMemoryTaskKiller, @ForQuery LowMemoryKiller lowMemoryQueryKiller)
```

Comment:
It's looks a little weird that we first need to bind available killers to a list and then inject a list. That makes the logic of killing (first tasks, then whole queries) not obvious and spread out across different places. Have you considered injecting `@ForTask LowMemoryKiller lowMemoryTaskKiller, @ForQuery LowMemoryKiller lowMemoryQueryKiller` into the `ClusterMemoryManager` directly?

Reply:
Crossed my mind - I agree it is slightly nicer. Will change.